### PR TITLE
Add a wiki search with a hotkey

### PR DESCRIPTION
**Pull request by Rottenbeer**

### CODE REVIEW

File: PTA/pta.cpp [hunk 1]
```diff
@@ -408,6 +456,59 @@ void PTA::handlePriceCheckHotkey(uint32_t flag)
...
+    keystroke.push_back(pta::CreateInput(VK_CONTROL, true));
+    keystroke.push_back(pta::CreateInput('C', true));
+    keystroke.push_back(pta::CreateInput('C', false));
+    keystroke.push_back(pta::CreateInput(VK_CONTROL, false));
+
+    SendInput(keystroke.size(), keystroke.data(), sizeof(keystroke[0]));
+
+    m_blockHotkeys = false;
+}
+
```
Comment (r52) [inline PTA/pta.cpp]: This doesn't look any different from `PTA::handlePriceCheckHotkey().` Should probably be removed entirely in favor of `PTA::handlePriceCheckHotkey()` and perhaps change that method's name to be more generic.

File: PTA/pta.cpp [hunk 2]
```diff
@@ -327,6 +363,18 @@ void PTA::setupFunctionality()
...
+	bool wikiEnabled = settings.value(PTA_CONFIG_WIKI_HOTKEY_ENABLED, true).toBool();
+
+    if (wikiEnabled)
+    {
+        QString seq = settings.value(PTA_CONFIG_WIKI_HOTKEY, PTA_CONFIG_DEFAULT_WIKI_HOTKEY).toString();
+
+        m_wikiKey.reset(new QHotkey(QKeySequence(seq), true));
+        qDebug() << "Wiki Search Hotkey Registered:" << m_wikiKey->isRegistered();
+
+        connect(m_wikiKey.get(), &QHotkey::activated, [=]() { handleWikiHotkey(WIKI_SEARCH); });
```
Comment (r52) [inline PTA/pta.cpp]: See comment for `PTA::handleWikiHotkey()`.

File: PTA/pta.h [hunk 3]
```diff
@@ -16,10 +16,11 @@ QT_FORWARD_DECLARE_CLASS(ItemAPI)
 
 struct PItem;
 
-enum price_check_flag : uint32_t
+enum search_check_flag : uint32_t
 {
     PC_SIMPLE = 0,
     PC_ADVANCED,
+	WIKI_SEARCH,
```
Comment (r52) [inline PTA/pta.h]: Please make sure that your editor is set to insert spaces and not tab characters for tabs.

File: PTA/pta.h [hunk 4]
```diff
@@ -68,6 +69,7 @@ private slots:
 
     void handleScrollHotkey(short data);
     void handlePriceCheckHotkey(uint32_t flag);
+    void handleWikiHotkey(uint32_t flag);
```
Comment (r52) [inline PTA/pta.h]: See comment for `PTA::handleWikiHotkey()`

File: PTA/pta_types.h [hunk 5]
```diff
@@ -20,6 +20,8 @@ constexpr auto PTA_CONFIG_SIMPLE_CHECK_HOTKEY         = "hotkey/simplecheck";
 constexpr auto PTA_CONFIG_SIMPLE_CHECK_HOTKEY_ENABLED = "hotkey/simplecheck/enabled";
 constexpr auto PTA_CONFIG_ADV_CHECK_HOTKEY            = "hotkey/advcheck";
 constexpr auto PTA_CONFIG_ADV_CHECK_HOTKEY_ENABLED    = "hotkey/advcheck/enabled";
+constexpr auto PTA_CONFIG_WIKI_HOTKEY				  = "hotkey/wiki";
+constexpr auto PTA_CONFIG_WIKI_HOTKEY_ENABLED		  = "hotkey/wiki/enabled";
```
Comment (r52) [inline PTA/pta_types.h]: Please ensure ClangFormat is executed on these changes.

**Comment by r52** [comment]
A couple of things to address as posted in the comments above, otherwise it looks good.

- Please make sure your Visual Studio editor is set to insert spaces instead of tabs (I use 4 spaces).
- Please ensure that you have [ClangFormat](https://i.imgur.com/Vh7MDYe.png) enabled in Visual Studio as well, and that you run code format before saving on the changes. A Visual Studio extension such as [Format document on Save](https://marketplace.visualstudio.com/items?itemName=mynkow.FormatdocumentonSave) can help you do this easily.

Thanks

File: PTA/pta.cpp (hunk 1 above)
Comment (Rottenbeer) [reply PTA/pta.cpp]: I removed handleWikiHotkey and renamed handlePriceCheckHotkey to handleItemHotkey to have a function that handles hotkey events with item interactions.

**Pull request merged**